src/olympia/scanners/models.py

```diff
@@ -120,6 +125,32 @@ class Meta:
     def __str__(self):
         return self.name
 
+    def clean(self):
+        if self.scanner == YARA:
+            self.clean_yara()
+
+    def clean_yara(self):
```

Comment:
For all your definition-related validation errors, you should do `raise ValidationError({'definition': ...})` instead so that the error is assigned to the definition field instead of being put with all the non-field errors at the top.

Reply:
Ha, right. Thanks! I filed https://github.com/mozilla/addons-server/issues/12910